features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/CategorySearchProvider.java

```diff
@@ -113,31 +108,30 @@ public boolean supportsPrefix(String searchPrefix) {
 
     @Override
     public void addVertexHopCriteria(SearchResult searchResult, GraphContainer container) {
-        CategoryHopCriteria criteria = m_categoryHopFactory.getCriteria(searchResult.getLabel());
+        CategoryHopCriteria criteria = createCriteria(searchResult.getLabel());
```

Comment:
Can the `createCriteria` method use `SearchResult` instead of `searchResult.getLabel()` instead?
The method could then also set the id.